# Dataset preparation details

[NeMu pipeline](nemu-pipeline.com) was used to derive mutational spectra

## Content

### Data generation code

- [Jupyter notebook](./prepare_beasts_datasets.ipynb) used to generate input protein sequences for pipeline execution
- [Used version of NeMu pipeline](./nemu.nf) for spectra calculation using protein sequences and nucleotide blast database (MIDORI2) as input
- [Config for the pipeline](./nemu_chordata.config)
- [Bash script](./run_chordata.sh) for parallel pipeline execution on around 10,000 mithochondrial genes of many different species

### Data aggregatrion and analysis

- [collect_spectra.ipynb](./collect_spectra.ipynb) - Jupyter notebook used to aggregate NeMu-pipeline outputs to simple tables with mutational spectra for the gene-species

### Final dataset

- [Mutational spectra (192-comp)](./dataset/MutSpecVertebrates192.csv.gz)
- [Input sequences info](./dataset/info.csv)

## Additional dependencies

We don't share too big external and raw files

- `prepare_beasts_datasets.ipynb` requires MIDORI2 database (https://www.reference-midori.info/, MIDORI2_UNIQ_NUC_GB259_{GeneName}_BLAST.fasta.zip)
- `collect_spectra.ipynb` requires raw spectra that can be generated by NeMu-pipeline (`nemu.nf` and `run_chordata.sh`)

## Materials for the article

Figures:
- [Supplementary Fig. S17](./figures/selected_genes_length.png)
- Supplementary Fig. S18 [a](./figures/mut_cnt_per_species.png) [b](./figures/tree_branch_mut_cnt.png) [c](./figures/ts_cnt_in_genes.png) [d](./figures/ts_frac_filter.png)

Tables:
- [Supplementary Table S1](./data/species_cnt_filtered.csv)
- [Supplementary Table S9](./dataset/info.csv)
- [Supplementary Table S10](./data/species_cnt_raw.csv)